src/syntax/parser.rs

```diff
@@ -376,7 +377,7 @@ impl<TokType: Tokenizer + Iterator<Item=Tokens>> Parser<TokType> {
 
         let expr = Expr::FnDecl(fn_name, args, return_type, definition);
 
-        return Some(ExprWrapper::default(expr));
+        Some(ExprWrapper::default(expr))
```

Comment:
its not at all obvious to me in that big match block below in your code, and because its a big, hundreds of lines long match block, if you're in the middle, you need to scroll and go look at the top or bottom of the match in order to know if its returning into the variable or function